transport/routing.go

```diff
@@ -0,0 +1,39 @@
+package transport
+
+import "math/rand"
+
+const (
+	// Range of ports that can be used to establish connection.
+	maxPort = 65535
+	minPort = 49152
+)
+
+type ShardInfo struct {
+	shard    uint16
+	nrShards uint16
+	// This will be used.
+	// msbIgnore uint8
+}
+
+// RandomShardPort returns randomly generated port that can be used
+// to establish connection to a specific shard on scylla node.
+func RandomShardPort(si ShardInfo) uint16 {
+	maxRange := int(maxPort - si.nrShards + 1)
+	minRange := int(minPort + si.nrShards - 1)
+	r := uint16(rand.Intn(maxRange-minRange+1) + minRange)
+	return r/si.nrShards*si.nrShards + si.shard
+}
+
+// ShardPortIterator returns iterator for consecutive ports that are
+// mapped to a specific shard on scylla node.
+func ShardPortIterator(si ShardInfo) func() uint16 {
+	curr := RandomShardPort(si)
+
+	return func() uint16 {
+		curr += si.nrShards
```

Comment:
cur?

Reply:
It was supposed to mean current. Will currentPort be more descriptive?